drivers/SmartThings/zigbee-power-meter/src/shinasystems/init.lua

```diff
@@ -74,8 +74,7 @@ local function energy_meter_handler(driver, device, value, zb_rx)
     device:set_field(constants.ENERGY_METER_OFFSET, offset, {persist = true})
   end
   raw_value_kilowatts = raw_value_kilowatts - offset
-  raw_value_kilowatts = math.floor((raw_value_kilowatts*10^3+0.5))/(10^3) -- Let's round it to 3 decimal(W) places.
```

Comment:
By removing this line (math.floor) and testing for four days, the strange graphs no longer appear.